pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -361,3 +365,34 @@ func (dsw *desiredStateOfWorld) GetPodToAdd() map[types.UniquePodName]PodToAdd {
 	}
 	return pods
 }
+
+func (dsw *desiredStateOfWorld) IsPodVolumePreviouslyAdded(podName volumetypes.UniquePodName, volumeSpec *volume.Spec, nodeName k8stypes.NodeName) (bool, error) {
```

Comment:
Change `volumeSpec *volume.Spec` to `volumeName v1.UniqueVolumeName` (consistent with `VolumeExists(...)`). And move the logic of extracting a volume name from the `volumeSpec` to the caller. Then you can also change the return for this to a simple bool (drop the error).